### PR TITLE
Another way to die - Could you please look whether this is a sensible way to go about this?

### PR DESCRIPTION
**Pull request by DavoudTaghawiNejad**
The problem was that replacing dead agents with dummies slowed down the simulation considerably. The new approach is to replace agents with None and, when new agents are created replace them. What do you think about this implementation?

### CODE REVIEW

File: abce/__init__.py [hunk 1]
```diff
@@ -535,10 +535,8 @@ def delete_agent(self, name, quite=True):
         group, id = name
 
         pg = self._processor_groups[id % self.processes]
-        if quite:
-            pg.replace_with_dead(group, id, SilentDeadAgent)
-        else:
-            pg.replace_with_dead(group, id, LoudDeadAgent)
+        self.num_of_agents_in_group[group] -= 1
+        pg.delete_agent(group, id)
```
Comment (rht) [inline abce/__init__.py]: At least the death should be logged at here.

File: abce/inventory.py [hunk 2]
```diff
@@ -3,14 +3,18 @@
 from abce.notenoughgoods import NotEnoughGoods
 from .expiringgood import ExpiringGood
 
-
 epsilon = get_epsilon()
 
 
+def isclose(a, b):
```
Comment (rht) [inline abce/inventory.py]: Yeah this is how it is implemented at elsewhere in the code.

Comment (DavoudTaghawiNejad) [reply abce/inventory.py]: what do you mean

File: abce/__init__.py (hunk 1 above)
Comment (DavoudTaghawiNejad) [reply abce/__init__.py]: fixed

**Comment by DavoudTaghawiNejad** [comment]
Update to give correct id if new agent is appended instead of replacing the slot of a dead agent.

**Pull request merged**